### PR TITLE
Batch indexing in ES.

### PR DESCRIPTION
**Pull request by manan164**
This PR contains changes to index as batch in Elastic Search V5.
1) Batch size can be determined by property workflow.elasticsearch.index.batchSize.
2) Added prometheus metrics to determine queue size and indexing time.
3) Unit test

@kishorebanala @apanicker-nflx @naveenchlsn  Please look at it.

### CODE REVIEW

File: es5-persistence/src/test/java/com/netflix/conductor/dao/es5/index/TestElasticSearchDAOV5.java [hunk 1]
```diff
@@ -372,6 +373,65 @@ public void indexTask() throws Exception {
...
+
+					assertTrue("should return 1 or more search results", result.getResults().size() > 0);
+					assertEquals("taskId should match the indexed task", "some-task-id", result.getResults().get(0));
+				});
+
+		embeddedElasticSearch.stop();
+		startElasticSearchWithBatchSize(1);
+	}
+
+	private void startElasticSearchWithBatchSize(int i) throws Exception {
```
[the remark this answers is not in the document]
Comment (manan164) [reply es5-persistence/src/test/java/com/netflix/conductor/dao/es5/index/TestElasticSearchDAOV5.java]: This actually restart the Embedded ES instance with the config as batch size 2.

File: es5-persistence/src/main/java/com/netflix/conductor/dao/es5/index/ElasticSearchRestDAOV5.java [hunk 2]
```diff
@@ -653,27 +661,40 @@ private void indexObject(final String index, final String docType, final String
...
+        }
     }
 
     /**
      * Performs an index operation with a retry.
      * @param request The index request that we want to perform.
      * @param operationDescription The type of operation that we are performing.
      */
-    private void indexWithRetry(final IndexRequest request, final String operationDescription) {
+    private synchronized void indexWithRetry(final BulkRequest request, final String operationDescription) {
```
Comment (manan164) [inline es5-persistence/src/main/java/com/netflix/conductor/dao/es5/index/ElasticSearchRestDAOV5.java]: @apanicker-nflx Sometime I see ConcurrentModificationException with List based approach. So making this method synchronized

File: core/src/main/java/com/netflix/conductor/metrics/Monitors.java [hunk 3]
```diff
@@ -267,4 +267,16 @@ public static void recordDaoError(String dao, String action) {
 	public static void recordAckTaskError(String taskType) {
 		counter(classQualifier, "task_ack_error", "taskType", taskType);
 	}
+
+	public static void recordESIndexTime(String docType, long val) {
+		getTimer(Monitors.classQualifier, docType, docType).record(val, TimeUnit.MILLISECONDS);
```
Comment (manan164) [inline core/src/main/java/com/netflix/conductor/metrics/Monitors.java]: I would like to ask two questions here,
1) We are using prometheus metrics. I see timer values as ```percentile="T0072",statistic="percentile"```. How to calculate 90 percentile value. As of now We are using totalTime / count to get average value.
2) How does  ```statistic="totalTime"``` calculated. Let say some task execution took 429ms. I see value as 4.29E9 although time unit says it is in seconds.
@apanicker-nflx @kishorebanala

Comment (apanicker-nflx) [reply core/src/main/java/com/netflix/conductor/metrics/Monitors.java]: @manan164 I'm not very familiar with Prometheus. We use [Atlas](https://github.com/Netflix/atlas) for collecting metrics.

**Pull request merged**